src/schema/schema.rs

```diff
@@ -358,6 +383,28 @@ impl Schema {
         }
         Ok(doc)
     }
+
+    /// Searches for a full_path in the schema, returning the field name and a JSON path.
+    ///
+    /// This function works by checking if the field exists for the exact given full_path.
+    /// If it's not, it splits the full_path at non-escaped '.' chars and tries to match the
+    /// prefix with the field names, favoring the longest field names.
+    ///
+    /// This does not check if field is a JSON field. It is possible for this functions to
```

Comment:
Would it make sense to improve the spec and avoid this?

Reply:
It might be. I changed the function to continue when the prefix is not a JSON field, and all of the tests passed (probably does not mean it is completely ok to change), but I think it might be better to open a separate PR for it.

Reply:
Can you open a ticket?

Reply:
#1699